### ISSUE TITLE
TypeError: Object #<Client> has no method 'disconnect'

### ISSUE THREAD

**Issue by Toddses** [mscdex/ssh2#228]
Seeing the following error when I try to connect to a bad host with code from the master branch.

```
Error: { [Error: Timed out while waiting for handshake] level: 'client-timeout' }

/var/www/apps/dploy/node_modules/ssh2/lib/client.js:136
      self.disconnect();
           ^
TypeError: Object #<Client> has no method 'disconnect'
    at null._onTimeout (/var/www/apps/dploy/node_modules/ssh2/lib/client.js:136:12)
    at Timer.listOnTimeout [as ontimeout] (timers.js:112:15)
```

I took a look at the referenced line (136) and tried to change `self.disconnect()` to `self.end()`, thinking we were just looking at a bad function call. Good news: that sort of solved the issue. When I change to `self.end()` it no longer produces the error, but the connection doesn't terminate either.

**Comment by mscdex** [comment]
Should be fixed in 4343af02c6a.

**Comment by Toddses** [comment]
yep. man you are just on this shit. i love it. thanks for the hard work.
